tests/test_naive_bayes.py

```diff
@@ -0,0 +1,394 @@
+	assert_almost_equal( multi.log_probability([ 3, 3 ]), -4.531024246969945 )
+	assert_almost_equal( multi.log_probability([ 1, 1 ]), -10.531024246969945 )
+	assert_almost_equal( multi.log_probability([ -1, -1 ]), -20.531024246969945 )
+
+	assert_almost_equal( indie.log_probability([ 11, 11 ]), -float('inf') )
+	assert_almost_equal( indie.log_probability([ 10, 10 ]), -4.605170185988091 )
+	assert_almost_equal( indie.log_probability([ 5, 5 ]), -4.605170185988091 )
+	assert_almost_equal( indie.log_probability([ 0, 0 ]), -4.605170185988091 )
+	assert_almost_equal( indie.log_probability([ -1, -1 ]), -float('inf') )
+
+	# create simple hmms, test values of hmms
+	rigged = State( DiscreteDistribution({ 'H': 0.8, 'T': 0.2 }) )
+	unrigged = State( DiscreteDistribution({ 'H': 0.5, 'T':0.5 }) )
+	
+	dumb = HiddenMarkovModel()
+	dumb.start = rigged
+	dumb.add_transition( rigged, rigged, 1 )
+	dumb.bake()
+
+	fair = HiddenMarkovModel()
+	fair.start = unrigged
+	fair.add_transition( unrigged, unrigged, 1 )
+	fair.bake()
+
+	smart = HiddenMarkovModel()
+	smart.add_transition( smart.start, unrigged, 0.5 )
+	smart.add_transition( smart.start, rigged, 0.5 )
+	smart.add_transition( rigged, rigged, 0.5 )
+	smart.add_transition( rigged, unrigged, 0.5 )
+	smart.add_transition( unrigged, rigged, 0.5 )
+	smart.add_transition( unrigged, unrigged, 0.5 )
+	smart.bake()
+
+	# make sure hmm's are working
+	assert_almost_equal( dumb.log_probability( list('H') ), -0.2231435513142097 )
+	assert_almost_equal( dumb.log_probability( list('T') ), -1.6094379124341003 )
+	assert_almost_equal( dumb.log_probability( list('HHHH') ), -0.8925742052568388 )
+	assert_almost_equal( dumb.log_probability( list('THHH') ), -2.2788685663767296 )
+	assert_almost_equal( dumb.log_probability( list('TTTT') ), -6.437751649736401 )
+	
+	assert_almost_equal( fair.log_probability( list('H') ), -0.6931471805599453 )
+	assert_almost_equal( fair.log_probability( list('T') ), -0.6931471805599453 )
+	assert_almost_equal( fair.log_probability( list('HHHH') ), -2.772588722239781 )
+	assert_almost_equal( fair.log_probability( list('THHH') ), -2.772588722239781 )
+	assert_almost_equal( fair.log_probability( list('TTTT') ), -2.772588722239781 )
+
+	assert_almost_equal( smart.log_probability( list('H') ), -0.43078291609245417 )
+	assert_almost_equal( smart.log_probability( list('T') ), -1.0498221244986776 )
+	assert_almost_equal( smart.log_probability( list('HHHH') ), -1.7231316643698167 )
+	assert_almost_equal( smart.log_probability( list('THHH') ), -2.3421708727760397 )
+	assert_almost_equal( smart.log_probability( list('TTTT') ), -4.1992884979947105 )
+	assert_almost_equal( smart.log_probability( list('THTHTHTHTHTH') ), -8.883630243546788 )
+	assert_almost_equal( smart.log_probability( list('THTHHHHHTHTH') ), -7.645551826734343 )
+
+	# to be used in each test
+	global univariate
+	global multivariate
+	global hmms
+
+	# no errors should be thrown
```

Comment:
have one setup for each of the naive bayes models you're going to use, and then several tests for each model